### PR TITLE
Add extra step to macos java install

### LINKED ISSUE #2814: Auto detecting java doesn't show javas in prisms java folder for macOS
### Operating System

macOS

### Version of Prism Launcher

9.0

### Version of Qt

6.7.2

### Description of bug

After downloading java, when I auto detect it shows other locations but doesn't show prism's java menu

### Steps to reproduce

1. Download java
2. auto detect
3. doesn't appear

### Suspected cause

_No response_

### This issue is unique

- [X] I have searched the issue tracker and did not find an issue describing my bug.

### PR DESCRIPTION
**Pull request by Trial97**
Parent PR: #2069
fixes #2814

so basically the java downloads on mac add another layer to java
this ensures that there is a link to the bin directory on the root java folder

### CODE REVIEW

**Review by Ryex — APPROVED** [review]

File: launcher/java/download/SymlinkTask.cpp [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+            return path;
+        }
+    }
+
+    return {};
+}
+
+void SymlinkTask::executeTask()
+{
+    setStatus(tr("Check Java bin"));
```
Comment (getchoo) [inline launcher/java/download/SymlinkTask.cpp]:
```suggestion
    setStatus(tr("Checking for Java binary path"));
```

File: launcher/java/download/SymlinkTask.cpp [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+    const auto binPath = FS::PathCombine("bin", "java");
+    const auto wantedPath = FS::PathCombine(m_path, binPath);
+    if (QFileInfo::exists(wantedPath)) {
+        emitSucceeded();
+        return;
+    }
+
+    setStatus(tr("Search for Java bin"));
```
Comment (getchoo) [inline launcher/java/download/SymlinkTask.cpp]:
```suggestion
    setStatus(tr("Searching for Java binary path"));
```

File: launcher/java/download/SymlinkTask.cpp [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+    const auto contentsPartialPath = FS::PathCombine("Contents", "Home", binPath);
+    const auto relativePathToBin = findBinPath(m_path, contentsPartialPath);
+    if (relativePathToBin.isEmpty()) {
+        emitFailed(tr("Failed to find bin java"));
```
Comment (getchoo) [inline launcher/java/download/SymlinkTask.cpp]:
```suggestion
        emitFailed(tr("Failed to find Java binary path"));
```

File: launcher/java/download/SymlinkTask.cpp [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+        return;
+    }
+    const auto folderToLink = relativePathToBin.chopped(binPath.length());
+
+    setStatus(tr("Collect folders to symlink"));
```
Comment (getchoo) [inline launcher/java/download/SymlinkTask.cpp]:
```suggestion
    setStatus(tr("Collecting folders to symlink"));
```

File: launcher/java/download/SymlinkTask.cpp [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+    auto entries = QDir(folderToLink).entryInfoList(QDir::NoDotAndDotDot | QDir::AllEntries);
+    QList<FS::LinkPair> files;
+    setProgress(0, entries.length());
+    for (auto& entry : entries) {
+        files.append({ entry.absoluteFilePath(), FS::PathCombine(m_path, entry.fileName()) });
+    }
+
+    setStatus(tr("Symlink Java"));
```
Comment (getchoo) [inline launcher/java/download/SymlinkTask.cpp]:
```suggestion
    setStatus(tr("Symlinking Java binary path"));
```

**Review by getchoo — CHANGES_REQUESTED** [review]

**Review by getchoo — APPROVED** [review]
LGTM. It would be nice to have the OP of the issue test this before merging, though

**Comment by getchoo** [comment]
This is also a follow up on the parent PR, so no need for a changelog entry imo

**Comment by Trial97** [comment]
> LGTM. It would be nice to have the OP of the issue test this before merging, though

We need to fix mac builds first but yeah I will ping OP about once they are merged

**Pull request merged**